src/libNode/FinalBlockProcessing.cpp

```diff
@@ -1927,3 +1970,60 @@ bool Node::ProcessForwardStateDelta(const vector<unsigned char>& message,
 
     return true;
 }
+
+void Node::CommitForwardedMsgBuffer()
+{
+    {
+        lock_guard<mutex> g(m_mutexForwardedTxnBuffer);
+
+        for (auto it = m_forwardedTxnBuffer.begin();
+             it != m_forwardedTxnBuffer.end();)
+        {
+            if (it->first < m_mediator.m_txBlockChain.GetLastBlock()
```

Comment:
Can this condition actually be hit?

Reply:
Yes, for the forwarding message too late to arrive

Reply:
I think the check for blocknum in `ProcessForwardTransaction` already takes care of this, but anyway no harm in keeping this code.